### ISSUE TITLE
"Invalid or missing message signature, please sign out and back in again" error on sending an email

### ISSUE THREAD

**Issue by itisafrickinhighlander** [vladimiry/ElectronMail#543]
Since upgrading to version 5.0.2 I started to have the issue (it has been so far intermittent, although it is in this state for about an hour since I noticed it so far) that I am unable to send any message and getting this message

`Invalid or missing message signature, please sign out and back in again`

I then tested PM web UI vs. EM and PM via web worked every time, while EM did not. I tested all 5.x.x releases and here's what I found:
- 5.0.0 works 
- 5.0.1 doesn't, but it is because it is, for whatever reason, throwing me `Internal server error`
- 5.0.2 doesn't with the heading issue.

I tested all these with a complete wipe of `%APPDATA%/electron-mail` between installations. I didn't use the local store, and I used the regular `proton.me` webstack.

I looked at the log (error-only, I enabled debug, but it was just a spam about notifications and subscriptions), here's what I get
```
[2022-08-11 10:13:19.794] [error] src\electron-main\web-contents.ts {"level":3,"message":"Uncaught (in promise) StatusCodeError","line":32264,"sourceId":"webclient://mail.proton.me/index.757b8173.js"}
[2022-08-11 10:13:19.795] [error] src\electron-main\web-contents.ts {"level":3,"message":"Uncaught (in promise) StatusCodeError","line":32264,"sourceId":"webclient://mail.proton.me/index.757b8173.js"}
[2022-08-11 10:13:19.798] [error] src\web\browser-window\app\_accounts\account-view.component.ts AccountViewComponent webview event {"type":"console-message","level":3,"message":"Uncaught (in promise) StatusCodeError","line":32264,"sourceId":"webclient://mail.proton.me/index.757b8173.js"}
[2022-08-11 10:13:19.798] [error] src\web\browser-window\app\_accounts\account-view.component.ts AccountViewComponent webview event {"type":"console-message","level":3,"message":"Uncaught (in promise) StatusCodeError","line":32264,"sourceId":"webclient://mail.proton.me/index.757b8173.js"}
```

**Comment by vladimiry** [comment]
I received a similar report about a week ago (via email). The same issue started to happen in 5.0.2, but rolling back to 5.0.1 was a workaround.

> since yesterday, i can only send to proton-addresses.
> when sending to any other addresses, it gives the error:
> Invalid or missing message signature, please sign out and back in again.

I tend to think it's about somehow broken proton web clients stack being packaged in the app as a static resource. Going to prepare a work-in-progress build with updated proton clients stack, and then we see if it helps.

**Comment by vladimiry** [comment]
> 5.0.0 works

Did you do re-login when rolled back to 5.0.0 (opened a new user session)?

**Comment by itisafrickinhighlander** [comment]
> > 5.0.0 works
> 
> Did you do re-login when rolled back to 5.0.0 (opened a new user session)?

Yes, I wiped `%APPDATA%/electron-mail` every time, so it was a fresh install per se.

**Comment by vladimiry** [comment]
> Yes, I wiped %APPDATA%/electron-mail every time, so it was a fresh install per se.

I was just curious if opened in 5.0.2/5.0.1 session would work without issues in 5.0.0 (the session opening `x-pm-appversion` header was the only major change apart from updated proton clients stack).

**Comment by itisafrickinhighlander** [comment]
I can confirm it, it does indeed work, I ran 5.0.0, updated to 5.0.2, the issue manifested. Then I downgraded back to 5.0.0 and it works again. All just by running the installer, no data were removed in between.

**Comment by vladimiry** [comment]
Thanks for helping to strike out the session-specific issue cause. So somehow broken proton clients stack remains a primary suspect.

**Comment by vladimiry** [comment]
> Going to prepare a work-in-progress build with updated proton clients stack, and then we see if it helps.

The currently deployed at proton.me mail web client version is 5.0.5.x/5.0.5.6. The v5.0.2 app release comes with 5.0.6.x/5.0.6.3 version. So I'm going to wait for proton.me updates mail web client to 5.0.6.x, and then I prepare the app build with the same client version.

**Comment by arch-btw** [comment]
@itisafrickinhighlander are you using tor? I'm having the same issue.

**Comment by itisafrickinhighlander** [comment]
> @itisafrickinhighlander are you using tor? I'm having the same issue.

I have been using the regular API, but I tried it via the Tor address and the issue persisted on both.

**Comment by vladimiry** [comment]
> So I'm going to wait for proton.me updates mail web client to 5.0.6.x

Ok, they deployed 5.0.6.8 https://mail.proton.me/assets/version.json. So I'm updating to the same client version and then prepare the app build for testing.

**Comment by vladimiry** [comment]
Can someone test [this](https://mega.nz/folder/MoBTTDLT#NOqoBaI4_b37QVwHw4A_yQ) build? It comes with 5.0.6.8 proton mail web client, same client version as currently deployed at proton.me.

**Comment by itisafrickinhighlander** [comment]
@vladimiry So far so good, sending seems to be working just fine. I will keep running it and in case anything pops up, I will let you know.

**Comment by vladimiry** [comment]
Thanks for the report. I'm going to wait for a few more days and then will likely be publishing a new release.

**Comment by vladimiry** [comment]
Should be fixed in v5.1.0.
